lineapy/graph_reader/session_artifacts.py

```diff
@@ -50,9 +56,16 @@ def __init__(
         self.artifact_nodecollections = []
         self.node_context = OrderedDict()
         self.input_parameters = input_parameters
+        self.use_cache = dict()
```

Comment:
lets name this differently from the input parameter, I think its cleanest to have the same name for the attribute and the inputto the constructor when you can do `self.x = x` in this case maybe rename this to `self.input_artifact_dict = dict()`